### DISCUSSION TITLE
[Suggest] a no_where() constraint for `DeleteStatement` and `UpdateStatement`

### DISCUSSION THREAD

**Discussion by Makiras** [SeaQL/sea-query#151] [Ideas]
Maybe `sea-query` can prevent the user update or delete whole table by mistake. That's **not necessary**, but humanize (for the junior coder like myself).

This is my draft ideas:
- force user call a function `.no_where()` when they are using `DeleteStatement` and `UpdateStatement` without `.xxx_where`.
- show warning/error when user complie the code.

I think this func will take a lot of time but no obvious benefit, so this is just a suggestion for small talk🍕.

**Comment by tyt2y3** [comment]
This might be a good idea, though has to be a breaking change indeed.

So the implementation strategy is, we deprecate the `build` method on `DeleteStatement` and a function with the signature

```rust
fn no_where(self) -> SafeDeleteStatement
```

that actually has a 'welcomed' `build` method
